chain/network/src/peer_manager.rs

```diff
@@ -357,6 +358,23 @@ impl PeerManagerActor {
             warn!(target: "network", "Unknown account {} in peers, not supported indirect routing", account_id);
         }
     }
+
+    fn process_announce_account(&mut self, peer_id: PeerId, account_id: AccountId, signature: Signature) -> AnnounceAccountResult {
+        if self.account_peers.contains_key(&account_id) {
+            // This announcement was already received.
+            return AnnounceAccountResult::IgnoreAnnounce;
+        }
+
+        // Verify the signature.
+        if !verify(account_id.as_bytes(), &signature, &peer_id.pub_key()) {
```

Comment:
we need to verify signature from ValidatorManager here, not from peer_id sadly.
ValidatorManager for each validator's account id contains special key that validators sign blocks - we need to verify that this peer, whatever their peer_id is (which is just some network id, ppl can decide to change any time), they are actually a validator in current or next epoch.